caffe2/CMakeLists.txt

```diff
@@ -256,12 +256,37 @@ endif()
 # ---[ Caffe2 HIP sources.
 if(USE_ROCM)
   # Call again since Caffe2_HIP_INCLUDES is extended with ATen include dirs.
-  IF(BUILD_ATEN)
-     HIP_INCLUDE_DIRECTORIES(${Caffe2_HIP_INCLUDES})
-  ENDIF()
-  IF(BUILD_CAFFE2)
-     set_source_files_properties(${Caffe2_HIP_SRCS} PROPERTIES HIP_SOURCE_PROPERTY_FORMAT 1)
-  ENDIF()
+  if(BUILD_ATEN)
+    get_directory_property(MY_DEFINITIONS COMPILE_DEFINITIONS)
```

Comment:
Give a brief comment about what's going on here, and also what a suitable way of generalizing this might be (because I suspect other people may want this code.)